### ISSUE TITLE
the model can not convergence when adding compactor

### ISSUE THREAD

**Issue by cutefulf** [DingXiaoH/ResRep#19]
Hi author, thank you for your grate work.
    I implemented the resrep pruner in tensorflow environment according to your paper and code. below is what  I did:
firstly I add compactors to the conv layers to be pruned. and then I restore the parameter from original model be to pruned. I confirmed the restored model have the same output with the original model.  I also implemented others works such as adding regularization to compactor's grad , mask  schedule, etc.
    after that  I begin to train the model, but I found the model can't  convergence, the loss is become bigger as the training on going( the mask is not set yet as the steps is not enough)，it seems like grad explosion？

did you meet the same issue before？

my environment：
tf2.2
network: mobilenetv2 + fpn + centernet. (the compactors only added to the mobilenetv2 backbone)

thank you！

**Comment by cutefulf** [comment]
optimizer: adam
lr schedule:
learning_rate: {
        manual_step_learning_rate {
          initial_learning_rate: 1e-3
          schedule {
           step: 80000
           learning_rate: 1e-4
          }
          schedule {
            step:120000
            learning_rate: 1e-5
          }
          schedule {
            step:160000
            learning_rate: 1e-6
          }
          schedule {
           step: 200000
           learning_rate: 1e-3
          }
    }
}

**Comment by cutefulf** [comment]
after I adjusted the initial lr from 1e-3 to 1e-4. the loss is normal. I think the cause seems like the  compactor may cause the model's error surface's landscape be rugged.
